lib/pos_consensus_controller.go

```diff
@@ -1,27 +1,193 @@
 package lib
 
 import (
-	"sync"
-
+	"github.com/deso-protocol/core/bls"
 	"github.com/deso-protocol/core/consensus"
+	"sync"
+	"time"
 )
 
 type ConsensusController struct {
 	lock                  sync.RWMutex
 	fastHotStuffEventLoop *consensus.FastHotStuffEventLoop
 	blockchain            *Blockchain
+	server                *Server
+
+	handshake *HandshakeController
```

Comment:
All of the changes in this file are related to peer management for the validator set, and related to maintaining connections between nodes. The changes introduced by this PR are not directly related to maintaining state of the blockchain or state of the Fast-HotStuff event loop based on internal events (block proposal, vote, timeout) and external messages from peers (blocks, vote, timeouts)

IMO, we should keep this ConsensusController scoped to just maintaining state blockchain and state of the Fast-HotStuff event loop based on the above events and and messages.

IMO, it's clearer to move all of the peer management, validator handshake, and validator connection maintenance from this file to a separate file, ex: `pos_validator_management.go` that's scoped to just these three items. The loop to re-sync the validator set may live in server or in the new file. IMO, this the ConsensusController is not the best place to house it though.